openassessment/xblock/static/js/spec/lms/oa_response.js

```diff
@@ -408,21 +432,51 @@ describe("OpenAssessment.ResponseView", function() {
-        view.prepareUpload(files);
+        var files = [{type: 'image/jpeg', size: 6000000, name: 'huge-picture.jpg', data: ''}];
+        view.prepareUpload(files, 'image');
         expect(baseView.toggleActionError).toHaveBeenCalledWith('upload', 'File size must be 5MB or less.');
     });
 
-    it("selects the wrong file type", function() {
+    it("selects the wrong image file type", function() {
         spyOn(baseView, 'toggleActionError').andCallThrough();
-        var files = [{type: 'bogus/jpg', size: 1024, name: 'picture.exe', data: ''}];
-        view.prepareUpload(files);
-        expect(baseView.toggleActionError).toHaveBeenCalledWith('upload', 'File must be an image.');
+        var files = [{type: 'image/jpg', size: 1024, name: 'picture.exe', data: ''}];
+        view.prepareUpload(files, 'image');
+        expect(baseView.toggleActionError).toHaveBeenCalledWith('upload', 'You can upload files with these file types: JPG, PNG or GIF');
+    });
+
+    it("selects the wrong pdf or image file type", function() {
+        spyOn(baseView, 'toggleActionError').andCallThrough();
+        var files = [{type: 'application/exe', size: 1024, name: 'application.exe', data: ''}];
+        view.prepareUpload(files, 'pdf-and-image');
+        expect(baseView.toggleActionError).toHaveBeenCalledWith('upload', 'You can upload files with these file types: JPG, PNG, GIF or PDF');
+    });
+
+    it("selects the wrong file extension", function() {
+        spyOn(baseView, 'toggleActionError').andCallThrough();
+        var files = [{type: 'application/exe', size: 1024, name: 'application.exe', data: ''}];
+        view.prepareUpload(files, 'custom');
+        expect(baseView.toggleActionError).toHaveBeenCalledWith('upload', 'You can upload files with these file types: pdf, doc, docx, html');
+    });
+
+    it("uploads an image using a one-time URL", function() {
+        var files = [{type: 'image/jpeg', size: 1024, name: 'picture.jpg', data: ''}];
+        view.prepareUpload(files, 'image');
+        view.fileUpload();
+        expect(fileUploader.uploadArgs.url).toEqual(FAKE_URL);
+        expect(fileUploader.uploadArgs.data).toEqual(files[0]);
+    });
+
+    it("uploads a PDF using a one-time URL", function() {
+        var files = [{type: 'application/pdf', size: 1024, name: 'application.pdf', data: ''}];
+        view.prepareUpload(files, 'pdf-and-image');
+        view.fileUpload();
+        expect(fileUploader.uploadArgs.url).toEqual(FAKE_URL);
+        expect(fileUploader.uploadArgs.data).toEqual(files[0]);
     });
 
-    it("uploads a file using a one-time URL", function() {
-        var files = [{type: 'image/jpg', size: 1024, name: 'picture.jpg', data: ''}];
-        view.prepareUpload(files);
+    it("uploads a arbitrary type file using a one-time URL", function() {
+        var files = [{type: 'text/html', size: 1024, name: 'index.html', data: ''}];
+        view.prepareUpload(files, 'custom');
```

Comment:
there is an test at line 456. I'll change the test name to be more clear on what we are testing.